byteman.yaml

```diff
@@ -0,0 +1,63 @@
+package:
+  name: byteman
+  version: 4.0.23
+  epoch: 0
+  description: Simplify Java tracing, monitoring and testing with Byteman
+  copyright:
+    - license: GPL-2.0-or-later AND BSD-3-Clause
+
+# NOTE: JRE runtime dependency required but only jar is needed at this point for using this package as part of hadoop.
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - zip
+      - unzip
+      - maven
+      - openjdk-9
+      - openjdk-9-default-jvm
+  environment:
+    JAVA_HOME: /usr/lib/jvm/java-1.9-openjdk
+    LANG: "en_US.UTF-8"
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/bytemanproject/byteman
+      tag: ${{package.version}}
+      expected-commit: 4e8bd4985ed3110ff0a2e226e154a72c0d7b8579
+
+  - runs: |
+      mvn clean install
+      mkdir -p ${{targets.destdir}}/usr/share/java/byteman
+      cp docs/copyright.txt ${{targets.destdir}}/usr/share/java/byteman/copyright.txt
+      cp licenses/third_party_licenses.txt ${{targets.destdir}}/usr/share/java/byteman/third_party_licenses.txt
+      mv byteman/target/byteman-${{package.version}}.jar ${{targets.destdir}}/usr/share/java/byteman/byteman.jar
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: "hadoop runner requires the jar file at /opt"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/opt/
+          ln -sf /usr/share/java/byteman/byteman.jar ${{targets.contextdir}}/opt/byteman.jar
+          ln -sf /usr/share/java/byteman/copyright.txt ${{targets.contextdir}}/opt/copyright.txt
+          ln -sf /usr/share/java/byteman/third_party_licenses.txt ${{targets.contextdir}}/opt/third_party_licenses.txt
+
+update:
+  enabled: true
+  github:
+    identifier: bytemanproject/byteman
+    use-tag: true
+
+test:
+  environment:
+    contents:
+      packages:
+        - openjdk-9
+        - openjdk-9-default-jvm
+  pipeline:
+    - name: jar listing
+      runs: |
+        jar -tf /usr/share/java/byteman/byteman.jar
```

Comment:
Is there a way to run the main class in the jar, or import it in a test and run a program that relies on it?

Reply:
Thanks Jamon, I've addressed this via https://github.com/wolfi-dev/os/pull/29349/commits/0bb32bc9479af2774d5aa9f552b3629dba62c5f1 (it runs a simple hello world app using byteman agent)

Reply:
Terrific, that's exactly the kind of test I was hoping for. Now we know it works as intended right at package build time.